api/lanes.py

```diff
@@ -1053,71 +1046,73 @@ def url_arguments(self):
+        """
+        facets = super(ContributorFacets, cls).from_request(
+            library, config, get_argument, get_header, worklist, *args,
+            **kwargs
+        )
+        facets.contributor = worklist.contributor
+        return facets
+
+    def navigate(self, *args, **kwargs):
+        """Propagate the selected contributor to a new Facets object."""
+        facets = super(ContributorFacets, self).navigate(*args, **kwargs)
+        facets.contributor = self.contributor
+        return facets
 
-class ContributorLane(WorksFromDatabase):
+    def modify_search_filter(self, filter):
+        super(ContributorFacets, self).modify_search_filter(filter)
+        filter.author = self.contributor
+
+
+class ContributorLane(DynamicLane):
     """A lane of Works written by a particular contributor"""
 
     ROUTE = 'contributor'
     MAX_CACHE_AGE = 48*60*60    # 48 hours
 
-    def __init__(self, library, contributor_name,
+    def __init__(self, library, contributor,
                  parent=None, languages=None, audiences=None):
-        if not contributor_name:
-            raise ValueError("ContributorLane can't be created without contributor")
+        """Constructor.
+
+        :param library: A Library.
+        :param contributor: A Contributor or ContributorData object.
+        :param parent: A WorkList.
+        :param languages: An extra restriction on the languages of Works.
+        :param audiences: An extra restriction on the audience for Works.
+        """
+        if not contributor:
+            raise ValueError(
+                "ContributorLane can't be created without contributor"
+            )
 
-        self.contributor_name = contributor_name
-        display_name = contributor_name
+        self.contributor = contributor
+        self.contributor_key = (
+            self.contributor.display_name or self.contributor.sort_name
+        )
         super(ContributorLane, self).initialize(
-            library, display_name=display_name,
+            library, display_name=self.contributor_key,
             audiences=audiences, languages=languages,
         )
         if parent:
             parent.append_child(self)
-        _db = Session.object_session(library)
-        self.contributors = _db.query(Contributor)\
-                .filter(or_(*self.contributor_name_clauses)).all()
 
     @property
     def url_arguments(self):
         kwargs = dict(
-            contributor_name=self.contributor_name,
+            contributor_name=self.contributor_key,
             languages=self.language_key,
             audiences=self.audience_key
         )
         return self.ROUTE, kwargs
 
-    @property
-    def contributor_name_clauses(self):
-        return [
-            Contributor.display_name==self.contributor_name,
-            Contributor.sort_name==self.contributor_name
-        ]
-
-    def apply_filters(self, _db, qu, facets, pagination, featured=False):
```

Comment:
Some of this code went into `ContributorData.lookup` and the rest went into `Filter.author_filter`.